### PR TITLE
Feature/posts structured data

### PR DESCRIPTION
**Pull request by EnriColombo**

### CODE REVIEW

File: app/Models/Event.php [hunk 1]
```diff
@@ -16,6 +18,7 @@ class Event extends Model implements HasMedia
     use HasFactory;
     use HasSlug;
     use InteractsWithMedia;
+    use HasStructuredData;
```
Comment (EnriColombo) [inline app/Models/Event.php:21]: Now use the new Trait HasStructuredData

File: app/Models/Post.php [hunk 2]
```diff
@@ -23,6 +25,7 @@ class Post extends Model implements HasMedia, Searchable
     use HasTranslations;
     use HasStatuses;
     use InteractsWithMedia;
+    use HasStructuredData;
```
Comment (EnriColombo) [inline app/Models/Post.php:28]: Post model implements HasStructuredData too...the idea is to add this Trait to each model we want it to generate tructured data for seo.

File: app/Traits/HasStructuredData.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * @package App\Traits
+ */
+trait HasStructuredData
+{
+    /**
+     * Factory method for generating the script for a Schema.org type.
+     *
+     * @return Type
+     */
+    abstract protected function generateStructuredDataScript() : Type;
```
Comment (EnriColombo) [inline app/Traits/HasStructuredData.php:25]: This is the abstract method each model has to implement (using spatie/schema-org library).

File: app/Traits/HasStructuredData.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * Render a json-ld script tag for the entity that implements this trait.
+     *
+     * @return string
+     */
+    public function toJsonLdScript() : string
+    {
+        return $this->generateStructuredDataScript()->toScript();
+    }
+}
```
Comment (EnriColombo) [inline app/Traits/HasStructuredData.php:36]: This method render the script in the json-ld format that you can use in a blade view.

File: resources/views/posts/show.blade.php [hunk 5]
```diff
@@ -36,6 +36,10 @@
 {{--    @endif--}}
 @endsection
 
+@section('structured-data')
+    {!! $post->toJsonLdScript() !!}
+@endsection
```
Comment (EnriColombo) [inline resources/views/posts/show.blade.php:41]: This is the call to the method of the model object to render the script.

**Review by davide-casiraghi — CHANGES_REQUESTED** [review]
The article title we need to duplicate it for 2 different schema.org properties.
- headline (you already did)
- name

**Review by davide-casiraghi — CHANGES_REQUESTED** [review]
Add about with the event category..
Same for the post with the post category.

**Review by davide-casiraghi — COMMENTED** [review]
Probably we can even use the subtype DanceEvent instead of Event..
Have a look to the event subtypes..
https://schema.org/Event

**Pull request merged**